src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -119,30 +127,35 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     IntermediateOutputPath:
     This is the full intermediate Output Path, and is derived from BaseIntermediateOutputPath, if none specified
-    (eg. obj\debug). If this property is overridden, then setting BaseIntermediateOutputPath has no effect.
+    (eg. obj\Debug). If this property is overridden, then setting BaseIntermediateOutputPath has no effect.
+
+    Ensure any and all path property has a trailing slash, so it can be concatenated.
     -->
 
   <PropertyGroup>
-    <!-- Ensure any OutputPath has a trailing slash, so it can be concatenated -->
-    <OutputPath Condition="'$(OutputPath)' != '' and !HasTrailingSlash('$(OutputPath)')">$(OutputPath)\</OutputPath>
+    <BaseOutputPath Condition=" '$(BaseOutputPath)' == '' ">bin\</BaseOutputPath>
+    <BaseOutputPath Condition="!HasTrailingSlash('$(BaseOutputPath)')">$(BaseOutputPath)\</BaseOutputPath>
+
     <AssemblyName Condition=" '$(AssemblyName)'=='' ">$(MSBuildProjectName)</AssemblyName>
     <!--
         Be careful not to give OutputPath a default value in the case of an invalid Configuration/Platform.
         We use OutputPath specifically to check for invalid configurations/platforms.
         -->
-    <OutputPath Condition=" '$(Platform)'=='' and '$(Configuration)'=='' and '$(OutputPath)'=='' ">bin\Debug\</OutputPath>
+    <OutputPath Condition=" '$(Platform)'=='' and '$(Configuration)'=='' and '$(OutputPath)'=='' ">$(BaseOutputPath)</OutputPath>
```

Comment:
I could append `Debug` there.

But do we have to? I mean, No `Configuration` or `Platform` is defined, so that means the build is happening without them, which might be single true build. So, isn't it better to place it in the root dir itself instead of creating a sub-dir?

Reply:
Whatever the best answer is, I'd be hesitant to change something that others can have relied on if, for instance, someone explicitly looks for `bin\Debug\...`, their code would now start failing where it hadn't previously.

Reply:
Exactly: don't change behavior of existing projects without a very good reason.

Reply:
I'll append `Debug` here.

It seems that Configuration is indeed set to Debug when empty.